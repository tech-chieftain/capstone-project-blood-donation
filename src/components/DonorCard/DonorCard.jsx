import React from 'react';
import { PropTypes } from 'prop-types';

function DonorCard({ title, cardIcon }) {
  const classesObjectStyles = {
    cardDonor: 'px-1  bg-gray-200 sm:w-full',
    cardIcon: 'w-6 h-6 px-1 my-3 mr-2',
    cardTitle: 'pt-3 text-lg sm:text-2xl md:text-2xl font-bold',
  };

  return (
    <div className={classesObjectStyles.cardDonor}>
      <div className="flex">
        <img className={classesObjectStyles.cardIcon} src={cardIcon} alt="icon" />
        <span className={classesObjectStyles.cardTitle}> {title} </span>
      </div>
    </div>
  );
}

DonorCard.propTypes = {
  title: PropTypes.element.isRequired,
  cardIcon: PropTypes.element,
};
export default DonorCard;
