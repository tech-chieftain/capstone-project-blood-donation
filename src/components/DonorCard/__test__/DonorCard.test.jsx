import React from 'react';
import renderer from 'react-test-renderer';
import DonorCard from '../DonorCard';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <DonorCard
        title="title"
        cardIcon="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
