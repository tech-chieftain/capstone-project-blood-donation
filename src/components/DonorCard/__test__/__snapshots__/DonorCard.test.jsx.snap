// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="px-1  bg-gray-200 sm:w-full"
>
  <div
    className="flex"
  >
    <img
      alt="icon"
      className="w-6 h-6 px-1 my-3 mr-2"
      src="https://apifortress.com/doc/wp-content/uploads/2019/04/ForgeIcon_Black.svg"
    />
    <span
      className="pt-3 text-lg sm:text-2xl md:text-2xl font-bold"
    >
       
      title
       
    </span>
  </div>
</div>
`;
